tests/integrationv2/test_record_padding.py

```diff
@@ -0,0 +1,160 @@
+    )
+
+    server_options = copy.copy(client_options)
+    server_options.data_to_send = None
+    server_options.mode = Provider.ServerMode
+    server_options.key = certificate.key
+    server_options.extra_flags = None
+
+    s2nd = managed_process(S2N, server_options, timeout=5)
+    openssl = managed_process(provider, client_options, timeout=5)
+
+    expected_total_length = None
+    if padding_size == 0:
+        # if padding size is zero, then the expected total length is
+        # equal to the payload size + 16 bytes of AEAD + 1 byte for content type
+        expected_total_length = PAYLOAD_SIZE + 16 + 1
+    else:
+        expected_total_length = padding_size + 16
+
+    expected_record_header = get_record_header(expected_total_length)
+
+    for client_results in openssl.get_results():
+        client_results.assert_success()
+        # verify that the openssl is sending padded payloads with the expected size
+        assert to_bytes(expected_record_header) in client_results.stdout
+
+    expected_version = get_expected_s2n_version(protocol, provider)
+
+    for server_results in s2nd.get_results():
+        server_results.assert_success()
+        # verify that the payload was correctly received by the server
+        assert random_bytes in server_results.stdout
+        # verify that the version was correctly negotiated
+        assert to_bytes("Actual protocol version: {}".format(
+            expected_version)) in server_results.stdout
+        # verify that the cipher was correctly negotiated
+        assert to_bytes("Cipher negotiated: {}".format(
+            cipher.name)) in server_results.stdout
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL])
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+# only tls 1.3 supports record padding
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", MINIMAL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("padding_size", PADDING_SIZES, ids=get_parameter_name)
+def test_s2n_13_client_handles_padded_records(managed_process, cipher, provider, curve, protocol, certificate,
+                                              padding_size):
+    port = next(available_ports)
+
+    random_bytes = data_bytes(PAYLOAD_SIZE)
+    server_options = ProviderOptions(
+        mode=Provider.ServerMode,
+        port=port,
+        cipher=cipher,
+        curve=curve,
+        cert=certificate.cert,
+        key=certificate.key,
+        insecure=True,
+        protocol=protocol,
+        # openssl errors when sending session tickets and when padding is set > 2075 bytes.
+        # use -rev s.t the openssl server echoes back the s2nc payload (but reversed)
+        extra_flags=['-num_tickets', 0,
+                     '-record_padding', padding_size, '-rev']
```

Comment:
If you want to get the server to send data to the client, you should use a send marker. It's more in-line with the rest of our tests than the -rev option.

Reply:
Actually, why do you even need to send extra messages? Doesn't Openssl pad handshake messages? Maybe you can just pull the header from those messages and not even bother with sending extra data.

Reply:
Fixed; discussed offline